src/Application.vala

```diff
@@ -32,6 +32,22 @@ public class ShortcutOverlay.Application : Gtk.Application {
 
         var main_window = new MainWindow (this);
         main_window.show_all ();
+
+        var quit_action = new SimpleAction ("quit", null);
+
+        add_action (quit_action);
+        add_accelerator ("Escape", "app.quit", null);
+
+        main_window.focus_out_event.connect ((event) => {
+            quit_action.activate (null);
+            return main_window.focus_out_event(event);
```

Comment:
This would work only if you had the access to the `base` of the main window. Here just tell Gtk that the event was handeled by returning `Gdk.EVENT_STOP`.